src/main/java/com/synopsys/integration/jenkins/detect/service/strategy/DetectScriptStrategy.java

```diff
@@ -82,21 +83,21 @@ public DetectScriptStrategy(JenkinsIntLogger logger, JenkinsProxyHelper jenkinsP
     }
 
     @Override
-    public MasterToSlaveCallable<String, IntegrationException> getSetupCallable() {
-        String scriptUrl;
+    public MasterToSlaveCallable<String, IntegrationException> getSetupCallable() throws IntegrationException {
```

Comment:
We probably don't want to throw an IntegrationException here-- if LATEST_POWERSHELL_SCRIPT_URL or LATEST_SHELL_SCRIPT_URL are not valid urls, that's problem we want to see at compile time, not runtime.

Reply:
How do we perform that type of check/validation at compile time?

Reply:
I've given this more thought and I no longer believe this is the wrong spot for this. We will get compile-time errors from DetectScriptStrategyCallableTest if the urls are invalid and that's probably good enough.